xgriddedaxis/remapper.py

```diff
@@ -1,3 +1,285 @@
+                fraction_overlap = (fromUB - toLB) / fromLength
+                weights.append(fraction_overlap * (fromLength / toLength))
+            elif (fromLB >= toLB) and (fromLB < toUB) and (fromUB >= toUB):
+                row_idx.append(r)
+                col_idx.append(c)
+                fraction_overlap = (toUB - fromLB) / fromLength
+                weights.append(fraction_overlap * (fromLength / toLength))
+            elif (fromLB >= toLB) and (fromUB <= toUB):
+                row_idx.append(r)
+                col_idx.append(c)
+                fraction_overlap = 1.0
+                weights.append(fraction_overlap * (fromLength / toLength))
+            elif (fromLB <= toLB) and (fromUB >= toUB):
+                row_idx.append(r)
+                col_idx.append(c)
+                fraction_overlap = (toUB - toLB) / fromLength
+                weights.append(fraction_overlap * (fromLength / toLength))
+
+    coverage = {
+        'weights': weights,
+        'col_idx': col_idx,
+        'row_idx': row_idx,
+        'shape': (m, n),
+    }
+    return coverage
+
+
+def construct_coverage_matrix(weights, col_idx, row_idx, shape, coords):
+    """
+    Generate remapping weights sparse matrix.
+
+    Parameters
+    ----------
+    weights : array_like
+        Contain overlap/coverage between the incoming and outgoing bounds
+    col_idx : array_like
+        column indices
+    row_idx : array_like
+        row indices
+    shape : tuple
+        Shape of the matrix
+    coords : dict
+        Dictionary-like container of coordinate arrays.
+
+    Returns
+    -------
+    xarray.DataArray
+        Contains the remapping weights (stored as a sparse matrix in COO format)
+    """
+    wgts = csr_matrix((weights, (row_idx, col_idx)), shape=shape).tolil()
+    mask = np.asarray(wgts.sum(axis=1)).flatten() == 0
+    wgts[mask, 0] = np.nan
+    wgts = sparse.COO.from_scipy_sparse(wgts)
+    weights = xr.DataArray(data=wgts, dims=['outgoing', 'incoming'], coords=coords)
+    return weights
+
+
+def _bounds_sanity_check(bounds):
+    # Make sure bounds is 2D
+    assert bounds.ndim == 2, f'Bounds must be a 2D array. Found bounds dimensions = {bounds.ndim}'
+    # Make sure bounds shape is of form (n, 2)
+    assert (
+        bounds.shape[1] == 2
+    ), f'Bounds must be a 2D array with shape: (n, 2). Found bounds shape = {bounds.shape}'
+    # Make sure lower_i <= upper_i
+    if np.any(bounds[:, 0] > bounds[:, 1]):
+        raise ValueError('all lower bounds must be smaller than their counter-part upper bounds')
+    # Make sure lower_i < lower_{i+1}
+    if np.any(bounds[0, :-1] >= bounds[0, 1:]):
+        raise ValueError('lower bound values must be monotonically increasing.')
+
+
+def _data_ticks_sanity_check(data_ticks):
+    assert data_ticks.ndim == 1, f'data ticks must be a one dimensional array.'
+    message = 'data ticks must be monotically increasing.'
+    assert np.all(data_ticks[:-1] <= data_ticks[1:]), message
```

Comment:
I actually don't think you can have `==` neighboring ticks.  So, this should be:

```suggestion
    assert np.all(data_ticks[:-1] < data_ticks[1:]), message
```

Reply:
Fixed!